benchmarks/lfds/hash_table.h

```diff
@@ -0,0 +1,118 @@
+#include <stdatomic.h>
+#include <assert.h>
+#include <stdbool.h>
+
+/*
+ * A simple lock-free hash table as described in
+ * https://preshing.com/20130605/the-worlds-simplest-lock-free-hash-table/
+ 
+ * Think of each thread as having its own private copy of the shared array. 
+ * Modifications to the shared array will eventually propagate to each 
+ * thread’s private copy, but not necessarily in the same order that they 
+ * were written. Therefore, when get() examines each array entry, its 
+ * private copy of that entry may contain any of the following combinations:
+ 
+ * 1. (0, 0)        The end of the item list. The key passed to get() was not found.
+ * 2. (key, 0)      An item that has not yet been fully initialized by a parallel call 
+ *                  to set(). If the key matches the one passed to get(), 0 is returned, 
+ *                  which is fine.
+ * 3. (key, value)  A fully initialized, valid item in the collection.
+ * 4. (0, value)    Only possible if the stores performed by a single set() call got 
+ *                  reordered either by the compiler or the processor, and have not yet fully 
+ *                  propagated to this thread’s private copy. get() will treat this as the 
+ *                  end of the item list, which means the key passed to get() was not found, 
+ *                  which is fine.
+ */
+
+#ifndef SIZE
+#define SIZE 2
+#endif
+_Static_assert ((SIZE & (SIZE - 1)) == 0, "SIZE must be a power of 2");
+
+struct Entry {
+    atomic_int key;
+    atomic_int value;
+};
+
+struct Entry m_entries[SIZE];
+
+inline static uint32_t integerHash(uint32_t h)
+{
+    // Original hash function
+	// h ^= h >> 16;
+	// h *= 0x85ebca6b;
+	// h ^= h >> 13;
+	// h *= 0xc2b2ae35;
+	// h ^= h >> 16;
+
+    // The above causes problems to the SMT solver.
+    // Since this does not affect correctness, we simply use identity.
+	return h;
+}
```

Comment:
It would be interesting to see how uninterpreted functions perform in this case.
Maybe once we reworked Dartagnan`s internals, it could be easy to just treat extern functions as uninterpreted (assuming no implementation is given).